design-documents/asynchronous-import/source-data-retrieving.md

```diff
@@ -0,0 +1,144 @@
+# Asynchronous Import Source data retrieving
+
+## Context
+
+The main idea is to receive data for import from different sources.
+
+## Requirements
+- Retrieving source data is encapsulated in a separate layer (including a separate module);
+- Ability to easily add new adapters for different data sources;
+- Ability to use streaming in different adapters;
+- Using only Magento API for source data retrieving;
+- Do not load whole data into memory;
+- Separation of the configuration (Part of the configuration should not be available for change through the admin panel).
```

Comment:
This may lead to bigger security issues than having configuration (creds to the source) provided with the source.
A user can provide just a link to the source, knowing that such file exists there, while not having permissions to that source.
IMO, creds must be passed together with the source.